pygbm/loss.py

```diff
@@ -156,6 +185,19 @@ def __call__(self, y_true, raw_predictions, average=True):
         return (logsumexp(raw_predictions, axis=1) -
                 (one_hot_true * raw_predictions).sum(axis=1))
 
+    def get_baseline_prediction(self, y_train, n_trees_per_iteration):
```

Comment:
I would rename `n_trees_per_iteration` as `prediction_dim` in the context of the loss function classes.

Reply:
I'll do it but I'm not sure it's always better. For example in `_update_gradients_hessians_categorical_crossentropy`, the update is semantically related to a tree, not to a 'dimension'.

Reply:
That's a good point. I was thinking that at some point we might want to try to implement multioutput regression trees such that we can have a categorical loss with one 1 tree per iteration. Anyways, we might rework that later.

Reply:
My point was to rename this local variable to `prediction_dim` only in the context of the loss function but keep the `n_trees_per_iteration` in the context of gradient boosting iteration.

But I think it's also ok to use the name `raw_prediction_dim` or `prediction_dim` everywhere.